src/app/proxy-receiver/service/proxyRequestHandler.ts

```diff
@@ -8,35 +8,43 @@ export enum ProxyRequestType {
 
 export type RequestListener = (request: Request) => void
 
-const ingressListeners = new Map<string, RequestListener>()
-const cdnListeners = new Map<string, RequestListener>()
+const ingressListeners = new Map<Key, RequestListener>()
+const cdnListeners = new Map<Key, RequestListener>()
+
+type Key = string & {
+  __keyBrand: never
+}
```

Comment:
I can't understand this type, what `__keyBrand`does and why does it have multiple underscores and why it is never?

Reply:
https://egghead.io/blog/using-branded-types-in-typescript

It's just an additional type-safety to make sure that we will pass a correct key to the functions.